### PR TITLE
Cleanup of header_rewrite redirect operator when invoked globally.

### PR DESCRIPTION
**Pull request by ywkaras**
Also includes new Au test coverage.

### CODE REVIEW

File: plugins/header_rewrite/operators.cc [hunk 1]
```diff
@@ -403,6 +403,14 @@ OperatorSetRedirect::initialize(Parser &p)
   require_resources(RSRC_RESPONSE_STATUS);
 }
 
+void
+OperatorSetRedirect::initialize_hooks()
+{
+  add_allowed_hook(TS_HTTP_READ_RESPONSE_HDR_HOOK);
+  add_allowed_hook(TS_HTTP_SEND_RESPONSE_HDR_HOOK);
+  add_allowed_hook(TS_REMAP_PSEUDO_HOOK);
```
Comment (ywkaras) [inline plugins/header_rewrite/operators.cc:411]: These are the only 3 hooks on which this operator actually works currently.  (By default, without a hook condition, operators execute on the READ_RESPONSE hook.)

**Comment by ywkaras** [comment]
[approve ci rocky]

File: plugins/header_rewrite/operators.cc [hunk 2]
```diff
@@ -430,35 +438,6 @@ EditRedirectResponse(TSHttpTxn txnp, const std::string &location, TSHttpStatus s
   TSHttpTxnErrorBodySet(txnp, TSstrdup(msg.c_str()), msg.length(), TSstrdup("text/html"));
 }
 
-static int
```
[the remark this answers is not in the document]
Comment (ywkaras) [reply plugins/header_rewrite/operators.cc:433]: It's no longer used.

File: tests/gold_tests/pluginTest/header_rewrite/header_rewrite_url_glob.test.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+tr.Processes.Default.StartBefore(ts)
+tr.Processes.Default.StartBefore(server)
+tr.Processes.Default.Command = (f'curl --head 127.0.0.1:{ts.Variables.port} -H "Host: 127.0.0.1:{server.Variables.Port}" --verbose')
+tr.Processes.Default.ReturnCode = 0
+tr.Processes.Default.Streams.stderr = "gold/set-redirect-glob.gold"
+tr.StillRunningAfter = server
+tr.StillRunningAfter = ts
+ts.Disk.traffic_out.Content = "gold/header_rewrite-tag.gold"
+
+tr = Test.AddTestRun()
```
[the remark this answers is not in the document]
Comment (ywkaras) [reply tests/gold_tests/pluginTest/header_rewrite/header_rewrite_url_glob.test.py:56]: Yes they are the same but meet different ID:REQUEST conditions.

[the remark this answers is not in the document]
Comment (ywkaras) [reply tests/gold_tests/pluginTest/header_rewrite/header_rewrite_url_glob.test.py:56]: I don' t understand.  You mean set `proxy.config.http.number_of_redirections` to non-zero?

[the remark this answers is not in the document]
Comment (jasmine-nahrain) [reply tests/gold_tests/pluginTest/header_rewrite/header_rewrite_url_glob.test.py:58]: I mean when the redirect returns an error code. Checking what the plugin does.

File: plugins/header_rewrite/operators.cc [hunk 4]
```diff
@@ -610,21 +589,9 @@ OperatorSetRedirect::exec(const Resources &res) const
       const_cast<Resources &>(res).changed_url = true;
       res._rri->redirect                       = 1;
     } else {
+      Dbg(pi_dbg_ctl, "OperatorSetRedirect::exec() hook=%d", int(get_hook()));
```
Comment (jasmine-nahrain) [inline plugins/header_rewrite/operators.cc:592]: is logging the int meaningful? Would it be more readable to have the string?

Comment (ywkaras) [reply plugins/header_rewrite/operators.cc:592]: I didn't have any trouble figuring it out, from looking at the code:  https://github.com/apache/trafficserver/blob/aaff5a9c922f4fbf1c3eacbf5864491d9cdd30c8/include/ts/apidefs.h.in#L450

File: tests/gold_tests/pluginTest/header_rewrite/header_rewrite_url_glob.test.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ywkaras) [reply tests/gold_tests/pluginTest/header_rewrite/header_rewrite_url_glob.test.py:58]: The plugin doesn't do the redirect.  If the configured number of redirections is 0 (the default), the client is expected to do redirect to the URL in the response Location header.  If redirections is greater than 0, maybe core ATS would do the redirection, but I'm not sure.

**Review by jasmine-nahrain — APPROVED** [review]

**Review by cmcfarlen — APPROVED** [review]

**Pull request merged**